hathor/transaction/storage/transaction_storage.py

```diff
@@ -958,6 +1024,7 @@ def get_transactions_that_became_invalid(self) -> List[BaseTransaction]:
         for tx in self.iter_mempool_from_best_index():
             if tx.is_spent_reward_locked():
                 tx.get_metadata().validation = ValidationState.INVALID
```

Comment:
Should we refactor it to `tx.set_validation(ValidationState.INVALID)`? I feel we've been calling `mark_partially_validated()` in several places.

Reply:
That would be good.

Reply:
Great refactor. Thanks!